### ISSUE TITLE
[Bug] Silent memory failure

### ISSUE THREAD

**Issue by unrealwill** [tensorflow/tensorflow#19155]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes (see below)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Lubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.7.0-3-g024aecf414 1.7.0
- **Python version**: Python 3.5.2
 **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**:  Driver Version: 390.30 CUDA Driver Version = 9.1
- **GPU model and memory**: NVidia GTX 1080Ti (11Go)
- **Exact command to reproduce**: copy paste the given code and call the only function (if running on a  GPU with different memory please fiddle with the size of the arrays accordingly)

### Describe the problem
When run on GPU the program runs fine except results are silently zeroed for elements in the last 25% of result. It only happens when I it uses a lot of memory but instead of failing it tries to compute and silently mess the results.

### Source code / logs

```
import tensorflow as tf
def bugTensorflow():
    #bug with GTX 1080 Ti
    #Lubuntu 16.04
    #Driver Version: 390.30
    #CUDA Driver Version = 9.1
    nbpoints1 = 100000
    nbpoints2 = 4000
    dim = 3

    #works fine on cpu and bugs on second gpu as well
    #with tf.device("/cpu:0"):
    #with tf.device("/gpu:1"):

    with tf.device("/gpu:0"):
        points = tf.random_normal( (nbpoints1,1,dim) )
        traj = tf.random_normal((1,nbpoints2,dim) )
        diff= points-traj
        dist = tf.reduce_sum( diff*diff,axis=-1)
    sess = tf.Session()
    with(sess.as_default()):
        _dist = sess.run(dist)
        print("dist.shape")
        print(dist.shape)
        print("dist[0]")
        print(_dist[0])
        print("dist[75000]")
        print(_dist[75000])
        print("tensorflow version")
        print(tf.__version__)
```
Results : 

> dist.shape
> (100000, 4000)
> dist[0]
> [ 3.918147   8.78706    3.0947132 ...  0.8662497  0.6857513 11.894537 ]
> dist[75000]
> [0. 0. 0. ... 0. 0. 0.]
> tensorflow version
> 1.7.0
>

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
Bazel version
Exact command to reproduce

**Comment by unrealwill** [comment]
@tensorflowbutler updated

**Comment by angersson** [comment]
@asimshankar Can you take a look at this?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @asimshankar, @angersson: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @asimshankar, @angersson: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by unrealwill** [comment]
@tensorflowbutler Damn humans are so lazy, you need to nagg harder :) otherwise they will never give you rights. You can start by CC'ing their bosses, or take money from them. That usually do the trick.

**Comment by chsigg** [comment]
Sorry for ignoring your issue. I've reproduced the issue and will provide updates in #22123.
